pkg/wharfapi/client.go

```diff
@@ -26,3 +33,38 @@ type Client struct {
 // Deprecated: This type has been renamed to Client and may be removed in a
 // future release.
 type WharfClient Client
+
+func (c Client) Get(from, path string, q url.Values) ([]byte, error) {
+	return doRequestNew(fmt.Sprintf("GET | %s", from), http.MethodGet, c.APIURL, path, q, nil, c.AuthHeader)
+}
+
+func (c Client) GetDecoded(response interface{}, from, path string, q url.Values) error {
+	bytes, err := c.Get(from, path, q)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(bytes, response)
+}
+
+func (c Client) Post(from, path string, q url.Values, body []byte) ([]byte, error) {
+	return doRequestNew(fmt.Sprintf("POST | %s", from), http.MethodPost, c.APIURL, path, q, body, c.AuthHeader)
+}
+
+func (c Client) PostDecoded(response interface{}, from, path string, q url.Values, body []byte) error {
```

Comment:
The method signatures here are a bit mixed.

As `PostDecoded` suggest a natural extension of `Post`, then the common parameters should come first.

I suggest:

```go
func (c Client) Post       (from, path string, q url.Values, body []byte) ([]byte, error) {}
func (c Client) PostDecoded(from, path string, q url.Values, body []byte, response interface{}) error {}
```

(spaces after `Post` are added just to make it clearer. You should not have those spaces in the code)